apps/examples/smartfs/read_optimization/read_optimization_main.c

```diff
@@ -0,0 +1,273 @@
+ * Name: create_test_file
+ *
+ * Description: Creates the test file with test data that we will use
+ *              to conduct the test.
+ *
+ ****************************************************************************/
+
+static int create_test_file(int f_size)
+{
+	int fd;
+	int ret;
+	memset(g_buf, '1', 1024);
+
+	fd = open(SAMPLE_FILE, O_WROK | O_CREAT);
+	if (fd < 0) {
+		printf("Unable to open Sample File for testing, Error = %d\n", errno);
+		return ERROR;
+	}
+
+	while (f_size >= 1024) {
+		ret = write(fd, g_buf, 1024);
+		if (ret != 1024) {
+			printf("Unable to write to Sample File, Error = %d\n", errno);
+			return ERROR;
+		}
+		f_size -= 1024;
+	}
+	if (f_size) {
+		ret = write(fd, g_buf, f_size);
+		if (ret != f_size) {
+			printf("Unable to write to Sample File, Error = %d\n", errno);
+			return ERROR;
+		}
+	}
+	close(fd);
+
+	return OK;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, char *argv[])
+#else
+int read_optimization_main(int argc, char *argv[])
+#endif
+{
+	int ret;
+	int size = 8;
+        int buf_size;
+	int frt_fd = -1;
+	int timer_devno = 0;
+	char timer_path[_POSIX_PATH_MAX];
+	struct timer_status_s time_1, time_2;
+	uint32_t time_read;
+	struct stat st;
+
+	if (argc > 1) {
+		timer_devno = atoi(argv[1]);
+		if (argc == 3) {
+			size = atoi(argv[2]);
+		}
+	} else {
+		printf("Correct Test Command Usage: smartfs_read_opt <Timer Device Number> <file Size in Kb>\n");
+		return -1;
+	}
+
+	snprintf(timer_path, _POSIX_PATH_MAX, TIMER_DEVNAME, timer_devno);
+	printf("Timer path = %s\n", timer_path);
+
+	ret = create_test_file(size * 1024);
+	if (ret != OK) {
+		printf("Unable to complete test\n");
+		return -1;
+	}
+
+	frt_fd = open(timer_path, O_RDONLY);
+	if (frt_fd < 0) {
+		fprintf(stderr, "ERROR: Failed to open Free Run Timer: %d\n", errno);
+		return -1;
+	}
+
+	if (ioctl(frt_fd, TCIOC_SETMODE, MODE_FREERUN) < 0) {
+		fprintf(stderr, "ERROR: Failed to set Free Run Timer: %d\n", errno);
+		return -1;
+	}
+	if (ioctl(frt_fd, TCIOC_START, TRUE) < 0) {
+		fprintf(stderr, "ERROR: Failed to start Free Run Timer: %d\n", errno);
+		return -1;
+	}
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_1) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	read_test(512);
+
+	if (ioctl(frt_fd, TCIOC_GETSTATUS, (unsigned long)(uintptr_t)&time_2) < 0) {
+		fprintf(stderr, "ERROR: Failed to get Free Run Timer status: %d\n", errno);
+		return -1;
+	}
+
+	time_read = time_2.timeleft - time_1.timeleft;
+	printf("\n****************************************************************************************\n");
+	printf("Time taken to read %dKb file with 512 byte buffer, 100 times = %lu\n", size, time_read);
```

Comment:
For better readability, could you include the throughput of each test, instead of the total time taken to read a file?

Reply:
Updated